server/tserver/src/main/java/org/apache/accumulo/tserver/TabletIteratorEnvironment.java

```diff
@@ -94,33 +94,22 @@ private TabletIteratorEnvironment(ServerContext context, IteratorScope scope,
     this.topLevelIterators = topLevelIterators;
   }
 
-  public TabletIteratorEnvironment(ServerContext context, IteratorScope scope,
-      AccumuloConfiguration config, ScanFileManager trm, Map<FileRef,DataFileValue> files,
-      Authorizations authorizations, SamplerConfigurationImpl samplerConfig) {
-    this(context, scope, config, trm, files, authorizations, samplerConfig, new ArrayList<>());
-  }
-
   public TabletIteratorEnvironment(ServerContext context, IteratorScope scope, boolean fullMajC,
-      AccumuloConfiguration config, MajorCompactionReason reason) {
+      MajorCompactionReason reason) {
     if (scope != IteratorScope.majc)
       throw new IllegalArgumentException(
           "Tried to set maj compaction type when scope was " + scope);
 
     this.context = context;
+    this.serviceEnvironment = new ServiceEnvironmentImpl(context);
     this.scope = scope;
     this.trm = null;
-    this.config = config;
     this.fullMajorCompaction = fullMajC;
     this.userCompaction = reason.equals(MajorCompactionReason.USER);
     this.authorizations = Authorizations.EMPTY;
     this.topLevelIterators = new ArrayList<>();
   }
 
-  @Override
-  public AccumuloConfiguration getConfig() {
```

Comment:
Will this just throw an exception now?

Reply:
Yeah I will revert this. I wasn't sure which impl was used on the server side when I was removing code.